### ISSUE TITLE
Integration of Aeotec switch gen5 (Model ZW075-C16)

### ISSUE THREAD

**Issue by sylvaincherrier** [OpenZWave/python-openzwave#78]
Hello,
i'm using a new zwave switch. The switch is integrated in my zwave network, and i get informations when i use api_demo.py. But this device is not recognized as a switch (get_switches() method says nothing)

here the result of get_switches() :
```
Nodes in network : 3
------------------------------------------------------------
Retrieve switches on the network
------------------------------------------------------------
------------------------------------------------------------
Retrieve dimmers on the network
```

here is the output of api_demo.py
```
4 - Name : 
4 - Manufacturer name / id : Unknown: id=0086 / 0x0086
4 - Product name / id / type : Unknown: type=0003, id=004b / 0x004b / 0x0003
4 - Version : 4
4 - Command classes : {'COMMAND_CLASS_HAIL', 'COMMAND_CLASS_DEVICE_RESET_LOCALLY', 'COMMAND_CLASS_NO_OPERATION', 'COMMAND_CLASS_ZWAVE_PLUS_INFO', 'COMMAND_CLASS_MANUFACTURER_SPECIFIC', 'COMMAND_CLASS_VERSION', 'COMMAND_CLASS_CRC_16_ENCAP'}
4 - Capabilities : {'beaming', 'listening', 'routing'}
4 - Neigbors : {1, 2}
4 - Can sleep : False
4 - Groups : {}
   ---------   
4 - Values for command class : COMMAND_CLASS_NO_OPERATION : {}
   ---------   
4 - Values for command class : COMMAND_CLASS_HAIL : {}
   ---------   
4 - Values for command class : COMMAND_CLASS_VERSION : {72057594119815191: {'readonly': True, 'max': 0, 'help': '', 'writeonly': False, 'label': 'Protocol Version', 'data_str': '3.83', 'min': 0, 'data': '3.83', 'ispolled': False, 'type': 'String', 'genre': 'System', 'units': ''}, 72057594119815207: {'readonly': True, 'max': 0, 'help': '', 'writeonly': False, 'label': 'Application Version', 'data_str': '3.26', 'min': 0, 'data': '3.26', 'ispolled': False, 'type': 'String', 'genre': 'System', 'units': ''}, 72057594119815175: {'readonly': True, 'max': 0, 'help': '', 'writeonly': False, 'label': 'Library Version', 'data_str': '3', 'min': 0, 'data': '3', 'ispolled': False, 'type': 'String', 'genre': 'System', 'units': ''}}
   ---------   
4 - Values for command class : COMMAND_CLASS_MANUFACTURER_SPECIFIC : {}
   ---------   
4 - Values for command class : COMMAND_CLASS_CRC_16_ENCAP : {}
   ---------   
4 - Values for command class : COMMAND_CLASS_DEVICE_RESET_LOCALLY : {}
   ---------   
4 - Values for command class : COMMAND_CLASS_ZWAVE_PLUS_INFO : {72057594119159809: {'readonly': True, 'max': 255, 'help': '', 'writeonly': False, 'label': 'ZWave+ Version', 'data_str': 1, 'min': 0, 'data': 1, 'ispolled': False, 'type': 'Byte', 'genre': 'System', 'units': ''}, 72057594119159830: {'readonly': True, 'max': 32767, 'help': '', 'writeonly': False, 'label': 'InstallerIcon', 'data_str': 1792, 'min': 4294934528, 'data': 1792, 'ispolled': False, 'type': 'Short', 'genre': 'System', 'units': ''}, 72057594119159846: {'readonly': True, 'max': 32767, 'help': '', 'writeonly': False, 'label': 'UserIcon', 'data_str': 1792, 'min': 4294934528, 'data': 1792, 'ispolled': False, 'type': 'Short', 'genre': 'System', 'units': ''}}
```

Thx

**Comment by bibi21000** [comment]
Your switch should be managed by openzwave : https://github.com/OpenZWave/open-zwave/blob/master/config/aeotec/zw075.xml
Did you include it in secure mode ?
Maybe you can delete your zwcfgxxx.xml and restart your app to build it again

**Comment by sylvaincherrier** [comment]
Thanks for your answer. Yes, it was included in secure mode. I think that maybe my problem is a silly one : where do you find the zwcfgxxxx.xml file ? On my raspberry, i have checked these files (i have several) but none at the correct date (i have excluded then re-included the faulty device. It works in z-way-server... but i can't find an update zwcfgxxxx.xml file)

**Comment by sylvaincherrier** [comment]
Thanks for your answer.
It was not perfectly clear for me how work the module, so i have tested it, and now, i understand.
the file zwcfgxxxx.xml is build at the first run of the module (it didn't get that).
so, i erased it, and start again.
I have the same error (the switch is not found by the get_switches() method.
The switch works well in z-way-server.
The only issue is that it is not proposed in get_switches() method.
i test with api_demo.py, but i made some modifications to run it with python3. I don't think it is the origin of the problem, as i have compared the content of the one created when using api_demo.py, and the one that is build when running the module in my app. they seem identical.

the generated file is here : 

[zwcfg_0xd536c579.xml.zip](https://github.com/OpenZWave/python-openzwave/files/995460/zwcfg_0xd536c579.xml.zip)

**Comment by bibi21000** [comment]
The lib uses command_classes to retrieve switches on your network : https://github.com/OpenZWave/python-openzwave/blob/master/src-api/openzwave/command.py#L454

The problem is that your node is not correctly detected : 

> <Manufacturer id="147" name="Unknown: id=0147"><Product type="400" id="2" name="Unknown: type=0400, id=0002"/></Manufacturer>

Looking at : https://github.com/OpenZWave/open-zwave/blob/master/config/manufacturer_specific.xml#L1097
It seems that it's not a switch.

**Comment by sylvaincherrier** [comment]
Hi ! Thx for your answer !
yes, that's strange, it should be :+1: https://github.com/OpenZWave/open-zwave/blob/master/config/manufacturer_specific.xml#L86  i think : Manufacturer 86, type 3 and id 4b

That is written in my zwcfgXXXX.xml file, and it is what i find in the corresponding file manufacturer_specific.xml at line 86.

And it is exactly my switch... so i think it is correctly detected, no ? and it works with the z-way-server... 
I'm gonna check one more time.

**Comment by bibi21000** [comment]
please attach your full configuration file

**Comment by sylvaincherrier** [comment]
Here it is
Python version : 3.4.2  (gcc 4.9.1)
lib version : 0.4.0.34


[zwcfg_0xd536c579.xml.zip](https://github.com/OpenZWave/python-openzwave/files/1022204/zwcfg_0xd536c579.xml.zip)

Thanks

**Comment by bibi21000** [comment]
Looking at your config.file; your switch is node with id 2, 
Looking at your first post, the config of the node 4 seems to be the one for node with id 5 in your config file.
Also, your nodes are node not detected correctly detected (ie product name). 
Could you check your user_path and config_path

**Comment by sylvaincherrier** [comment]
Hi,
the difference between the two posts is, i think, because i did some tests and erase the device 4 from the config to re-associate it (i wanted to verify if i was in secure mode).
Maybe the issue is linked to config_path and user_path... I have installed python_openzwave through pip3. and i use a modified version of api_demo.py to test it (modified because i use python3).
I have tried to display config_path and user_path. The answers are . and .
and when i check with pyozw_check, it tells me that config_path is _/usr/local/lib/python3.4/dist-packages/python_openzwave/ozw_config_ , which seems correct...

but when i run api_demo, it seems to use another version : 
/usr/local/lib/python3.4/dist-packages/openzwave/option.py

(i don't remember what i did on this raspberry, maybe i was using a self installed version of python-openzwave, then i have installed python-openzwave using pip3...)

This could explain that i have problems, while a user with a fresh install has no problem ?

**Comment by sylvaincherrier** [comment]
Solved.
I have restarted from a fresh install.
had to install libudev-dev
pip3 install python-openzwave
reconfiguration of my zwave devices.
then, using a modified version of api_demo (for python3) : 
the switch is displayed in -> _Retrieve switches all compatibles devices on the network_

The issue was link to the upgrade from the manual installation to the pip3 install.
[zwcfg.xml.zip](https://github.com/OpenZWave/python-openzwave/files/1038889/zwcfg.xml.zip)
